src/core/container.rs

```diff
@@ -69,7 +69,11 @@ where
     /// This method does **not** magically expose the given port, it simply performs a mapping on
     /// the already exposed ports. If a docker image does not expose a port, this method will not
     /// be able to resolve it.
-    pub fn get_host_port(&self, internal_port: u16) -> Option<u16> {
+    pub fn get_host_port<T>(&self, internal_port: T) -> Option<u16>
+    where
+        T: Into<u16> + Copy,
```

Comment:
If this is only `Into<u16>`, how are we going to map a non-TCP port? As far as I understand, the user needs to able to specify the protocol here.

Reply:
We might need to create our own trait here due to coherence rules.

Reply:
The `Into<u16>` is misleading here, it only is necessary for the debug print, I think it would be better to switch to `{:?}` there. The next line is the important one, because it let's us forward the call to the type specific implementation of `map_to_host_port`. You can try it in this Playground example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=100daefbe4f085548bfc3c620f522b82
I should even change the return value into `Option<T>`, so that the user couldn't accidentally mix up different port types - missed that here, because it was a bit late as a have done the commit.